tests/test_transform.py

```diff
@@ -0,0 +1,187 @@
+"""
+LSST Data Management System
+See COPYRIGHT file at the top of the source tree.
+
+This product includes software developed by the
+LSST Project (http://www.lsst.org/).
+
+This program is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+GNU General Public License for more details.
+
+You should have received a copy of the LSST License Statement and
+the GNU General Public License along with this program. If not,
+see <http://www.lsstcorp.org/LegalNotices/>.
+"""
+from __future__ import absolute_import, division, print_function
+import unittest
+
+import numpy as np
+import astshim
+
+import lsst.utils.tests
+import lsst.afw.geom as afwGeom
+
+def makeRawArrayData(nPoints, nAxes, delta=0.123):
+    return np.array([[i + j*delta for j in range(nAxes)] for i in range(nPoints)])
+
+def makeRawPointData(nAxes, delta=0.123):
+    return [i*delta for i in range(nAxes)]
+
+# names of enpoints
+NameList = ("Generic", "Point2", "Point3", "SpherePoint")
+
+def getNaxes(name):
+    """For a given endpoint name, return a tuple containing reasonable #s of axes
+
+    Note that only Generic has more than one possible number of axes
+    """
+    return {
+        "Generic": (1, 2, 3, 4, 5),
+        "Point3": (3,),
+        "Point2": (2,),
+        "SpherePoint": (2,),
+    }[name]
+
+def getBadNAxes(name):
```

Comment:
These two methods seem a bit excessive, since only Generic has more than one element, and it has no possible "bad" values. It would be simpler to me to just define a dict for each and use that, unless you plan to add other endpoint types.